elfpy/pricing_models/hyperdrive.py

```diff
@@ -257,39 +262,48 @@ def calc_out_given_in(
         TradeResult
             The result of performing the trade.
         """
+
+        # pylint: disable=too-many-locals
+
         # Calculate some common values up front
         in_amount = Decimal(in_.amount)
-        normalized_time = Decimal(time_remaining.normalized_time)
+        curve_portion = Decimal(time_remaining.normalized_time)
+        flat_portion = 1 - curve_portion
         share_price = Decimal(market_state.share_price)
-        d_bonds = in_amount * (1 - normalized_time)
+        d_bonds = in_amount * flat_portion  # whether in_.unit is base or pt, at maturity d_bonds = d_base
         d_shares = d_bonds / share_price
         # Redeem the matured bonds 1:1 and simulate these updates hitting the reserves.
-        market_state = market_state.copy()  # don't want to modify the actual market state
+        market_state_simulated = market_state.copy()  # don't want to modify the actual market state
         if in_.unit == types.TokenType.BASE:
-            market_state.share_reserves += float(d_shares)
-            market_state.bond_reserves -= float(d_bonds)
+            market_state_simulated.share_reserves += float(d_shares)
+            market_state_simulated.bond_reserves -= float(d_bonds)
         elif in_.unit == types.TokenType.PT:
-            market_state.share_reserves -= float(d_shares)
-            market_state.bond_reserves += float(d_bonds)
+            market_state_simulated.share_reserves -= float(d_shares)
+            market_state_simulated.bond_reserves += float(d_bonds)
         else:
             raise AssertionError(
                 "pricing_models.calc_out_given_in: ERROR: "
-                f"Expected in_.unit to be {types.TokenType.BASE} or {types.TokenType.PT}, not {in_.unit}!"
+                f"Expected in_.unit to be {types.TokenType.PT}, not {in_.unit}!"
```

Comment:
What was wrong with the old message? BASE or PT are allowed.

Reply:
yeah i forgot to undo this change, changing back